From da3cfbd8ee3585ddcb655ec0d7fcf06ac86bbe94 Mon Sep 17 00:00:00 2001
From: Jonas Knutsson <jonas5.knutsson@sonymobile.com>
Date: Tue, 12 Nov 2013 16:20:04 +0100
Subject: [PATCH 05/12] Display model name on exchange server and exclude
 illegal characters

All devices register on exchange server will list on server side
and show as "Android" as device name.
Devices brand and model needs to display on exchange server to make
better information. Most operators has requirement to show device
brand and model for device name, such as one of Canada's
largest telecommunications company Bell mobile did.

Devices brand and model needs to display on exchange server according
to most operator's requirement. Exchange server only accepts the device
type which contains letter and digital. That means if a device type
contains such as "-",Exchange server will reject this device.
So trim all other characters.

Change-Id: I0efd8cba8f08f625b950d31a38a0367697c8639d
Aosp: https://android-review.googlesource.com/#/c/99579
---
 src/com/android/exchange/service/EasServerConnection.java | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/com/android/exchange/service/EasServerConnection.java b/src/com/android/exchange/service/EasServerConnection.java
index 87d5b284..be4e0ed6 100644
--- a/src/com/android/exchange/service/EasServerConnection.java
+++ b/src/com/android/exchange/service/EasServerConnection.java
@@ -79,7 +79,11 @@ public class EasServerConnection {
      */
     protected static final long COMMAND_TIMEOUT = 30 * DateUtils.SECOND_IN_MILLIS;
 
-    private static final String DEVICE_TYPE = "Android";
+    // Exchange server only accepts letter and digital as device type. Trim all
+    // other characters.
+    private static final String DEVICE_TYPE =
+            (Build.BRAND + Build.MODEL).replaceAll("[^a-zA-Z0-9]", "");
+
     private static final String USER_AGENT = DEVICE_TYPE + '/' + Build.VERSION.RELEASE + '-' +
         Eas.CLIENT_VERSION;
 
-- 
2.17.1

